### ISSUE TITLE
OverlayFragment is sometimes added multiple times 

### ISSUE THREAD

**Issue by ro-mak** [pandulapeter/beagle#93]
Hi!
There is a chance that in UiManagerContract multiple OverlayFragments are added, or maybe some of them are not removed properly due to asynchronous transactions or state losses
Checking that activity.supportFragmentManager.fragments has OverlayFragment before adding a new one seems to help with the issue but I'd recommend to check it or find a better solution
```
 if (activity.supportFragmentManager.fragments.lastOrNull() !is OverlayFragment) {
      var isFragmentFound = false
                    activity.supportFragmentManager.fragments.forEach {
                        if (it is OverlayFragment) {
                            isFragmentFound = true
                        }
                    }
                    if (!isFragmentFound) {
                    activity.supportFragmentManager
                        .beginTransaction()
                        .run {
                          //  findOverlayFragment(activity)?.let(::remove)
                            add(android.R.id.content, OverlayFragment.newInstance())
                        }
                        .setReorderingAllowed(true)
                        .commitAllowingStateLoss()
                }
}
```

**Comment by pandulapeter** [comment]
Hi!

Thanks for reporting the issue. This is something that keeps popping up from time to time, I did attempt to fix it multiple times (thus the ugly code in the class you mentioned) but with the many ways people are using Fragments I was never sure that it's fixed. Fortunately it does not seem to cause significant issues (except for the unnecessary performance overhead in debug builds).

Let me do some testing with the solution you proposed, I'll let you know how it goes.

**Comment by pandulapeter** [comment]
Shoul be fixed in [v2.9.1](https://github.com/pandulapeter/beagle/releases)
